README.md

```diff
@@ -14,6 +13,61 @@ The template will then expect the source code to be located in the solutions-[re
 ## CF template and Lambda function
 Located in deployment/dist
 
+## Citizens Advice modifications and changes to the build process
+Any changes to the lambdas should be made to the code in
+```bash
+source/
+```
+Any changes to the template that generates waf.rb and waf.yml should be made to 
+```bash
+deployment/aws-waf-security-automations.template
+```
+If changes are made to the code of lambdas for which `deployment/aws-waf-security-automations.template` currently references `"%%BUCKET_NAME%%"` for the lambda code, the reference to `"%%BUCKET_NAME%%"` in the template file needs to be changed to `"Ref": "LambdaBucket"` to ensure that the changes get picked up when waf tiers are deployed.
+
+If changes are made to the file structure of any lambdas, then the `./build-s3-dist.sh` file needs to be amended so that the lambda code gets copied to the correct locations.
+
+**Process for re-creating waf.rb and waf.yml from aws code**
+
+1. Publish any customised lambda code to the cab S3 bucket where the template is expecting to find it, and tell the template to look for any non-customised code in the aws `solutions-eu-west-1` bucket (this will replace `%%BUCKET_NAME%%` in a copy of the template saved in deployment/dist and the `eu-west-1` gets added automatically):
+
+```bash
+cd deployment
+./build-s3-dist.sh solutions
+aws s3 cp dist/ s3://cab-shared-global-storage-waf-lambdas-170181076180/aws-waf-security-automations --recursive --profile cab-prod
+```
+
+2. Create waf.rb and waf.yml from the template in deployment/dist:
+
+```bash
+cd dist
+cp aws-waf-security-automations.template ../../../devops-tools/aws.template
+cd ../../../devops-tools
+be cfn stack to_dsl aws.template # creates aws.rb and aws.yml file in /infrastructure/cfn/cloudformation/templates
+cd infrastructure/cfn/cloudformation/templates 
+rubocop -a aws.rb # clean .rb file
+mv aws.rb waf/waf.rb # will be prompted to overwrite existing waf.rb...
+mv aws.yml waf/waf.yml # ...and waf.yml
+```
+
+3. No changes are needed to waf.rb, but need to ensure correct parameters in waf.yml. Surround each yes/no answer in single quotes 'yes'/'no', and set the following parameter values:
```

Comment:
You should be able to avoid this by changing the defaults value for the parameters in the CloudFormation template instead.

Reply:
Not sure I can get around the quotes bit (known issue that others have raised), and don't think I can set the stack output ones either, since the to_dsl function won't know how to handle it? So I'm tempted to leave all the changes to the yml file in one place?